### DISCUSSION TITLE
How to correctly use Alchemy provider with transport function in createWalletClient

### DISCUSSION THREAD

**Discussion by tx-nikola** [wevm/viem#2514] [Question]
When trying to use the Alchemy RPC, we tried creating the wallet client like this:
```
const [account] = await window.ethereum!.request({ method: "eth_requestAccounts" });

const walletClient = createWalletClient({
  account,
  chain: zkSync,
  transport: fallback
  ([ 
    http('https://zksync-mainnet.g.alchemy.com/v2/<API_KEY>'), 
    http('https://eth-mainnet.g.alchemy.com/v2/<API_KEY>'),
  ]),
}).extend(eip712WalletActions());
```

Initially we just tested with zkSync network, but later added the eth fallback as well, but didn't help.

The issue might also be with how Alchemy designed the API - [here is one example](https://docs.alchemy.com/reference/eth-accounts-zksync) where we can see that `eth_accounts` method exists on eth mainnet, but not on zksync.

This means that if we use the Alchemy RPC for zkSync, and we try to call some methods like
```
const [account] = await this.wallet.getAddresses();
```
it will fail, as the `getAddresses` method is calling an unsupported RPC method under the hood.
![image](https://github.com/user-attachments/assets/cc9115c6-263d-4712-b8d8-c2dec4b1c4cd)
![image](https://github.com/user-attachments/assets/a5f130a3-5de2-4a04-b161-116d7d2b1559)

Is there something we're not understanding correctly, or if we're doing something wrong?

Our main use case is that we need to call the `writeContract` method on the zkSync network, but the same issue happens.
When I try to use writeContract, it tries to call the "`eth_signTypedData_v4`" RPC method and sends the request to the alchemy endpoint, which doesn't have this method, thus the request fails (returns 400).

**Comment by jordaniza** [comment]
We also ran into this problem at Aragon, solution for now was to directly use window.ethereum but it would be nice to use our existing RPC infra
